internal/graph/graph.go

```diff
@@ -50,6 +50,13 @@ type ResolutionMetadata struct {
 	// evaluated and potentially discarded
 	// If the solution is "allowed=false", no paths were found. This is the sum of all the reads in all the paths that had to be evaluated
 	DatastoreQueryCount uint32
+
+	//Temporary for testing PR
+	TMP_Singleflight struct {
+		NumInnerCalls    int
+		NumOuterCalls    int
+		HadSharedRequest bool
+	}
```

Comment:
Not actually proposed code. Just serves as rudimentary metrics to benchmark vs other resolvers.